src/pixelator/pixeldataset/precomputed_layouts.py

```diff
@@ -253,81 +387,94 @@ def data():
         return PreComputedLayouts.create_empty()
 
 
-# TODO The code below this point is not yet tested, and should be considered
-# experimental. It is likely to change in the future. It will be exposed as
-# public to this method in the future.
-
-
 def _zero_fill_missing_markers(dataframe: pd.DataFrame, all_markers) -> pd.DataFrame:
     missing_markers = all_markers - set(dataframe.columns.to_list())
     for marker in missing_markers:
         dataframe[marker] = 0
     return dataframe
 
 
-def _get_layout(
+def _compute_layouts(
     edgelist: pl.DataFrame,
-    add_node_marker_counts,
-    layout_algorithm,
-    all_markers,
+    add_node_marker_counts: bool,
+    layout_algorithms: list[SupportedLayoutAlgorithm] | SupportedLayoutAlgorithm,
+    all_markers: list[str],
 ):
+    if isinstance(layout_algorithms, str):
+        layout_algorithms = [layout_algorithms]
+
     def data():
         for component, df in edgelist.group_by("component"):
             logger.debug("Computing for component %s", component)
-            yield (
-                pl.DataFrame(
-                    Graph.from_edgelist(
-                        df.lazy(),
-                        add_marker_counts=add_node_marker_counts,
-                        simplify=True,
-                        use_full_bipartite=True,
+            graph = Graph.from_edgelist(
+                df.lazy(),
+                add_marker_counts=add_node_marker_counts,
+                simplify=True,
+                use_full_bipartite=True,
+            )
+            for layout_algorithm in layout_algorithms:
+                yield (
+                    pl.DataFrame(
+                        graph.layout_coordinates(
+                            layout_algorithm=layout_algorithm,
+                            get_node_marker_matrix=add_node_marker_counts,
+                            cache=False,
+                        )
+                        .pipe(
+                            _zero_fill_missing_markers
+                            # If we don't want to add the marker counts
+                            # we just pass this through an identity function
+                            if add_node_marker_counts
+                            else lambda df, all_markers: df,
+                            all_markers=all_markers,
+                        )
```

Comment:
Interesting expression. That'd be cool to understand. Or simplify if people are going to be same interpretation at I do.

Reply:
This is basically equivalent with the psudo-code below:

```
def f(df: pd.DataFrame, add_node_marker_counts: bool):
   if add_node_marker_counts:
       return _zero_fill_missing_markers(df)
   return df
```
But written with https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.pipe.html I think it mad sense here since in general we are just chaining functions on each other here.